week7/answers.sql

```diff
@@ -1,3 +1,256 @@
+
+    -- calc the damage
+    SET damage = damage - armor; -- substracting armor from damage
+    IF damage < 0 THEN
+        SET damage = 0; -- this is assuming we don't care about the durability of the armor
+    END IF;				      -- or it regenerates on it's onw like in Call of Duty	
+
+    -- set the health
+    SELECT health INTO health FROM character_stats WHERE character_id = id_of_character_being_attacked;
+
+    -- Sub the damage
+    SET health = health - damage; -- I'm comparing health with the damage from before
+
+	-- now we update the health or delete the poor victim of database violence 
+    IF health > 0 THEN
+        UPDATE character_stats SET health = health WHERE character_id = id_of_character_being_attacked;
+    ELSE
+        DELETE FROM character_stats WHERE character_id = id_of_character_being_attacked;
+        DELETE FROM characters WHERE character_id = id_of_character_being_attacked; -- He's dead! T_T
+        DELETE FROM inventory WHERE character_id = id_of_character_being_attacked;
+        DELETE FROM equipped WHERE character_id = id_of_character_being_attacked;
+        DELETE FROM team_members WHERE character_id = id_of_character_being_attacked;
+    END IF;
+END ;;
+DELIMITER ;
+
+DELIMITER ;;
+CREATE PROCEDURE equip(IN inventory_id INT)
+BEGIN
+    DECLARE character_id INT;
+    DECLARE item_id INT;
+
+    -- getting id's from the inventory... Final Fantasy didn't work like I thought it did
+SELECT character_id, item_id INTO character_id, item_id 
+    FROM inventory AS inv 
+    WHERE inv.inventory_id = inventory_id; -- I aliased this because it was confusing me too much
+										                       -- and the hamster complained
+    -- del from inventory
+    DELETE FROM inventory AS inv 
+    WHERE inv.inventory_id = inventory_id;
+
+    -- Insert the item into the equipped table
+    INSERT INTO equipped (character_id, item_id) VALUES (character_id, item_id);
+END ;;
+DELIMITER ;
+
+DELIMITER ;;
+CREATE PROCEDURE unequip(IN equipped_id INT)
+BEGIN
+    DECLARE character_id INT;
+    DECLARE item_id INT;
+
+    -- find id's from equipped table
+    SELECT character_id, item_id INTO character_id, item_id 
+    FROM equipped AS eqp 
+    WHERE eqp.equipped_id = equipped_id;
+
+    -- Delete item
+    DELETE FROM equipped AS eqp 
+    WHERE eqp.equipped_id = equipped_id;
+
+    -- Insert to inventory
+    INSERT INTO inventory (character_id, item_id) 
+    VALUES (character_id, item_id);
+END ;;
+DELIMITER ;
+
+DELIMITER ;; -- I think this is what you meant in canvas(Update the winners 
+			 -- table so that only the characters in the passed team on in the winners table.)
+CREATE PROCEDURE set_winners(IN team_id INT)
+BEGIN
+    -- Delete the whole stuff in here
+    DELETE FROM winners;
+
+    -- Insert from the team passed as argument to winners... This is what I understood you wanted
+    INSERT INTO winners (character_id, name)
```

Comment:
Logically, this is correct; however, the goal of this part of the assignment is to use a CURSOR (we never covered nested SELECT statements, which don't scale to larger datasets).